### ISSUE TITLE
Adaptive numbers style based on length/pagination

### ISSUE THREAD

**Issue by xerus2000** [sharkdp/bat#1334]
Hey, I am really enjoying `bat` for peeking at everything, but here is one little pet peeve I recently encountered:
I only want line numbers when the auto-pagination sets in. Beforehand, the files are so small that it doesn't matter, and having no numbers makes copying much easier. But in the pager, they are super useful!

I tried to create a workaround by using bat as a pager for itself, but something doesn't seem to work here:
```sh
❯ bat --style header --pager "bat --style numbers" "/tmp"
There is no style option ("less --help" for help)
numbers: No such file or directory
```

**Comment by sharkdp** [comment]
> I tried to create a workaround by using bat as a pager for itself, but something doesn't seem to work here:

Ahh, we disallow usage of `bat` as a pager in order to avoid infinite recursion. It seems like the implementation is currently less than ideal and also buggy. Instead of printing a warning ("You can not use `bat` as a pager for `bat`…"), we silently replace `bat` by `less` and keep all command-line arguments (which doesn't make sense, because they are obviously not applicable to `less`). This should be fixed.

As for your use case: `bat` *always* calls the pager (unless specifically configured otherwise), even for small files. The pager then decides to quit itself if the file is small enough. At that point, we can not control whether or not to show line-numbers anymore. I think this would have to be solved in a separate wrapper script.

**Comment by xerus2000** [comment]
Oh wait, `less` does that? Uh.
But what does `--pagination auto` (which seems to be the default) even do then?
```sh
       --paging <when>

              Specify  when  to use the pager. To disable the pager, use '--paging=never' or its alias, -P. To disable the pager permanently, set BAT_PAGER to an empty string. To con‐
              trol which pager is used, see the '--pager' option. Possible values: *auto*, never, always.
```

**Comment by xerus2000** [comment]
> I think this would have to be solved in a separate wrapper script.

So I have to first cat the file, measures its length (`wc -l`) and then pipe it into bat with the appropriate arguments? Considering that I also want the bat header style, this doesn't sound great tbh - I'd have to access the file at least twice in what I came up with.

**Comment by sharkdp** [comment]
Removing the "bug" label, as #1343 has been merged.



> So I have to first cat the file, measures its length (`wc -l`) and then pipe it into bat with the appropriate arguments? Considering that I also want the bat header style, this doesn't sound great tbh - I'd have to access the file at least twice in what I came up with.

It's not great, no. But I'm not really sure we can (or want) to add support for this in `bat` directly. `bat` works in a line-oriented way and does not buffer results. This means that it can work in a streaming context (e.g. `some-program | bat -l json`). So we can not really decide to switch to another style once we have printed more than `$screen_height` lines.
